lib/agent0/agent0/hyperdrive/interactive/i_hyperdrive.py

```diff
@@ -0,0 +1,532 @@
+        if policy_config is not None and policy_config.rng is None and policy_config.rng_seed is None:
+            policy_config.rng = self.config.rng
+        out_agent = IHyperdriveAgent(
+            pool=self,
+            policy=policy,
+            policy_config=policy_config,
+            private_key=private_key,
+        )
+        return out_agent
+
+    def _init_agent(
+        self,
+        policy: Type[HyperdriveBasePolicy] | None,
+        policy_config: HyperdriveBasePolicy.Config | None,
+        private_key: str,
+    ):
+        # Setting the budget to 0 here, we'll update the wallet from the chain
+        agent = HyperdriveAgent(
+            Account().from_key(private_key),
+            initial_budget=FixedPoint(0),
+            policy=IHyperdrivePolicy(
+                IHyperdrivePolicy.Config(sub_policy=policy, sub_policy_config=policy_config, rng=self.config.rng)
+            ),
+        )
+
+        # TODO add the public address to the chain object to avoid multiple objects
+        # with the same underlying account
+        self.chain._ensure_no_duplicate_addrs(agent.checksum_address)  # pylint: disable=protected-access
+
+        agent.wallet = build_wallet_positions_from_chain(
+            agent.checksum_address, self.interface.hyperdrive_contract, self.interface.base_token_contract
+        )
+        return agent
+
+    def _set_max_approval(self, agent: HyperdriveAgent):
+        # Establish max approval for the hyperdrive contract
+        asyncio.run(
+            set_max_approval(
+                [agent],
+                self.interface.web3,
+                self.interface.base_token_contract,
+                str(self.interface.hyperdrive_contract.address),
+            )
+        )
+
+    def _add_funds(
+        self, agent: HyperdriveAgent, base: FixedPoint, eth: FixedPoint, signer_account: LocalAccount | None = None
+    ) -> None:
+        # The signer of the mint transaction defaults to the agent itself, unless specified.
+        if signer_account is None:
+            signer_account = agent
+        if eth > FixedPoint(0):
+            # Eth is a set balance call
+            eth_balance, _ = self.interface.get_eth_base_balances(agent)
+            new_eth_balance = eth_balance + eth
+            _ = set_anvil_account_balance(self.interface.web3, agent.address, new_eth_balance.scaled_value)
+
+        if base > FixedPoint(0):
+            # We mint base
+            _ = smart_contract_transact(
+                self.interface.web3,
+                self.interface.base_token_contract,
+                signer_account,
+                "mint(address,uint256)",
+                agent.checksum_address,
+                base.scaled_value,
+            )
+            # Update the agent's wallet balance
+            agent.wallet.balance.amount += base
+
+    # TODO this should be the base agent class for these calls
```

Comment:
do you mean the agent type should be the base agent class? I don't think that's right -- open_long as an action assumes hyperdrive.

Reply:
if that's not what you meant can you clarify in the comment?